### PR TITLE
feat: Additional Pester tests for AVM folder structure checks

### PR DESCRIPTION
**Pull request by eriqua**
## Description

Add dedicated Pester `Context` for Top level module folder tests:
- Move test 
  - Module should contain a [` version.json `] file
- Add tests:
  - Module should contain a [` tests `] folder
  - Module should contain a [` tests/e2e `] folder
  - Module should contain a [` tests/e2e/*waf-aligned `] folder
  - Module should contain a [` tests/e2e/*defaults `] folder
  - Module should contain one [` main.test.bicep `] file in each e2e test folder
  - Module should contain a [` tests/unit `] folder
  - Module should contain a [` tests/unit/custom.tests.ps1 `] file

Depending on #585 

[avm.res.network.private-endpoint](https://github.com/eriqua/bicep-registry-modules/actions/workflows/avm.res.network.private-endpoint.yml)
[avm.res.key-vault.vault](https://github.com/eriqua/bicep-registry-modules/actions/workflows/avm.res.key-vault.vault.yml)
[avm.res.insights.action-group](https://github.com/eriqua/bicep-registry-modules/actions/workflows/avm.res.insights.action-group.yml) (missing tests/unit folder and placeholder, solved by #585 )

### CODE REVIEW

File: avm/utilities/pipelines/staticValidation/compliance/module.tests.ps1 [hunk 1]
```diff
@@ -113,35 +121,58 @@ Describe 'File/folder tests' -Tag 'Modules' {
...
-      $moduleTestFilePaths = Get-ModuleTestFileList -ModulePath $moduleFolderPath | ForEach-Object { Join-Path $moduleFolderPath $_ }
-      $moduleTestFilePaths.Count | Should -BeGreaterThan 0
+      $e2eTestFolderPathList = Get-ChildItem -Directory (Join-Path -Path $moduleFolderPath 'tests' 'e2e')
+      foreach ($e2eTestFolderPath in $e2eTestFolderPathList) {
+        $pathExisting = Test-Path (Join-Path -Path $e2eTestFolderPath 'main.test.bicep')
+        $pathExisting | Should -Be $true
+      }
+    }
+
+    It '[<moduleFolderName>] Module should contain a [` tests/unit `] folder.' -TestCases $topLevelModuleTestCases {
```
Comment (AlexanderSehr) [inline avm/utilities/pipelines/staticValidation/compliance/module.tests.ps1:158]: I'd veto this one (or at least would like to put it up to debate, but the rest looks great, thanks :)

File: avm/utilities/pipelines/staticValidation/compliance/module.tests.ps1 [hunk 2]
```diff
@@ -113,35 +121,58 @@ Describe 'File/folder tests' -Tag 'Modules' {
...
+
+      param(
+        [string] $moduleFolderPath
+      )
+
+      $pathExisting = Test-Path (Join-Path -Path $moduleFolderPath 'tests' 'unit')
+      $pathExisting | Should -Be $true
+    }
+
+    It '[<moduleFolderName>] Module should contain a [` tests/unit/custom.tests.ps1 `] file.' -TestCases $topLevelModuleTestCases {
```
Comment (AlexanderSehr) [inline avm/utilities/pipelines/staticValidation/compliance/module.tests.ps1:168]: That's actually not a requirement. 'Any' Pester test file in the Unit folder would be picked up :) The name doesn't matter. One could argue though - if the test folder exists, it should contain at least 1 `.test.ps1` file

File: avm/utilities/pipelines/staticValidation/compliance/module.tests.ps1 (hunk 1 above)
Comment (eriqua) [reply avm/utilities/pipelines/staticValidation/compliance/module.tests.ps1:158]: I'm also not sure about this and that's the main reason for keeping this in draft. I can remove the 2 tests checking for the unit folder and files, and have them introduced again if we ever decide for that. The remaining tests are more urgent anyway

Comment (eriqua) [reply avm/utilities/pipelines/staticValidation/compliance/module.tests.ps1]: Removed

File: avm/utilities/pipelines/staticValidation/compliance/module.tests.ps1 (hunk 2 above)
Comment (eriqua) [reply avm/utilities/pipelines/staticValidation/compliance/module.tests.ps1]: Removed

**Review by AlexanderSehr — APPROVED** [review]

**Pull request merged**